src/gd.c

```diff
@@ -2339,8 +2342,8 @@ BGD_DECLARE(void) gdImageRectangle (gdImagePtr im, int x1, int y1, int x2, int y
 	}
 }
 
-BGD_DECLARE(void) gdImageFilledRectangle (gdImagePtr im, int x1, int y1, int x2, int y2,
-		int color)
+static void _gdImageFilledRectangle (gdImagePtr im, int x1, int y1, int x2, int y2,
+		int color, int horizontal)
```

Comment:
I'm not sure if _gdFilledPatternRectangle() would be a good choice, because this function actually draws *all* filled rectangles (the original gdImageFilledRectangle() being only a small wrapper on top of it). What about _gdImageFilledRectangleImp() or maybe _drawFilledRectangle()?

Renaming the last parameter to direction or axis is certainly more clarifying if we use an enum (or simple constants) instead of a bool. Not sure, if there's already something useful in libgd what can be reused – will have a look.

Another option would be to have two functions, say _gdImageFilledHRectangle() and _gdImageFilledVRectangle() instead. That would suit nicely to [gdImageHLine()](https://github.com/libgd/libgd/blob/gd-2.2.1/src/gd.c#L1247) and [gdImageVLine()](https://github.com/libgd/libgd/blob/gd-2.2.1/src/gd.c#L1266), for instance. The drawback would be yet more code duplication unless we'd use a macro (function inlining might not be available?)